custom_model_runner/datarobot_drum/drum/gpu_predictors/base.py

```diff
@@ -87,6 +87,12 @@ def __init__(self):
         except ImportError:
             raise DrumCommonException("OpenAI Python SDK is not installed")
 
+    def _supports_chat(self):
+        return True
+
+    def _chat(self, completion_create_params):
+        return self.ai_client.chat.completions.create(**completion_create_params)
+
```

Comment:
Is this really all it takes? That would be cool if so. Does additional work need to be done to support monitoring or is that out-of-scope?

FYI, we currently don't support users to hook into any `predict()`, `transform()`, etc traditional hooks. The only hook we support is the `load_model()` hook. I think for this first iteration we don't need to support a the `chat()` hook in `custom.py` unless if you feel differently.

Reply:
Yes, the heavy lifting is done in `BaseLanguagePredictor` so this is all it takes. Monitoring also works out of the box.

>  I think for this first iteration we don't need to support a the chat() hook in custom.py unless if you feel differently.

Sounds good to me 👍 

I'll just add a chat request to the [functional test](https://github.com/datarobot/datarobot-user-models/blob/eafb54079755ba4b1297315c08bc29d4423fe97b/tests/functional/test_inference_per_framework.py#L1183), and then this PR should be ready.

Reply:
Does this work for Triton too?

Reply:
No. Triton has its own API.

Reply:
@elatt Looks like the GPU functional tests are not running on PR pipelines? How do you usually run them? I don't know if we have access to GPUs on harness?

Reply:
yeah, I don't believe we have any GPU infrastructure in Harness 😞 (cc @carsongee @andrey-hura ?). So Alex and I each have our own GPU dev machines in AWS that we spin up and down anytime we need to do GPU related work / run the test.

Reply:
If you push your changes up to this branch, I can try running them (potentially).

Reply:
We can do special infra in Harness quite simply: https://datarobot.atlassian.net/wiki/spaces/CIIT/pages/5643272267/Harness+Special+Infrastructure . I'm not sure we have GPU specifically configured yet. cc @nullspoon . We'd need to work through the budgetary concerns for this to make sure we have a clear use case and budget prior to turning it on, but not hard from a technical perspective. Write me up an expected annual cost and business value real quick and we can add it. cc @timnolan1

Reply:
@elatt I have added chat to the test so would be great if you can try it out.

Reply:
I'm hitting this (which I don't think I've hit before
```python
2024-10-08 17:26:21,836 ERROR drum:  
Traceback (most recent call last):
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot_drum/drum/main.py", line 118, in main
    runtime.cm_runner.run()
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot_drum/drum/drum.py", line 485, in run
    self._run_predictions_pipelines_in_mlpiper()
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot_drum/drum/drum.py", line 854, in _run_predictions_pipelines_in_mlpiper
    self._pipeline_executor.init_pipeline()
  File "/workspace/datarobot/lib/python3.10/site-packages/mlpiper/pipeline/executor.py", line 276, in init_pipeline
    self._init_pipeline()
  File "/workspace/datarobot/lib/python3.10/site-packages/mlpiper/pipeline/executor.py", line 229, in _init_pipeline
    self._dag.configure_connected_pipeline(
  File "/workspace/datarobot/lib/python3.10/site-packages/mlpiper/pipeline/dag.py", line 118, in configure_connected_pipeline
    dag_node.component_runner.configure(input_args)
  File "/workspace/datarobot/lib/python3.10/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py", line 10, in configure
    self._dag_node.main_cls().configure(params)
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot_drum/resource/components/Python/prediction_server/prediction_server.py", line 128, in configure
    self._predictor.mlpiper_configure(params)
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot_drum/drum/gpu_predictors/base.py", line 112, in mlpiper_configure
    super().mlpiper_configure(params)
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot_drum/drum/language_predictors/base_language_predictor.py", line 106, in mlpiper_configure
    self._init_mlops()
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot_drum/drum/language_predictors/base_language_predictor.py", line 132, in _init_mlops
    self._mlops.init()
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot/mlops/mlops.py", line 129, in init
    self._model = Model(feature_types=self._feature_types)
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot/mlops/model.py", line 65, in __init__
    self._report_queue = OutputChannelQueueSync()
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot/mlops/channel/output_channel_queue.py", line 362, in __init__
    self._output_channel.open()
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot/mlops/spooler/api_spooler.py", line 67, in open
    self.set_config()
  File "/workspace/datarobot/lib/python3.10/site-packages/datarobot/mlops/spooler/api_spooler.py", line 52, in set_config
    raise DRSpoolerException(f"Configuration values missing: {missing}")
datarobot.mlops.common.exception.DRSpoolerException: Configuration values missing: [ConfigKey(name='DATAROBOT_ENDPOINT', type=<class 'str'>), ConfigKey(name='DATAROBOT_API_TOKEN', type=<class 'str'>)]
```

Do you know if MLOps reporting has become required now -- for running tests like this it would be great if there was an option to just bypass reporting.

Reply:
Ah this is the difference
```python
def _should_enable_mlops(self):
        return to_bool(self._params.get("monitor")) or self._supports_chat()
```
Since we support chat, we are not force to fake it out in this test 😞

Reply:
This should have been fixed in #1134 which I merged earlier today. MLOps should not be enabled [if deployment id is not set](https://github.com/datarobot/datarobot-user-models/blob/ac80235ab6b8340fbbc5b7f54fd50e4de1597e0e/custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py#L124)